common/djangoapps/util/db.py

```diff
@@ -4,44 +4,119 @@
 from functools import wraps
 import random
 
-from django.db import connection, transaction
+from django.db import DEFAULT_DB_ALIAS, DatabaseError, Error, transaction
 
 
 MYSQL_MAX_INT = (2 ** 31) - 1
 
 
-def commit_on_success_with_read_committed(func):  # pylint: disable=invalid-name
+class CommitOnSuccessManager(object):
```

Comment:
This is mostly taken from https://github.com/django/django/blob/1.8.5/django/db/transaction.py#L110-L277 but instead of using save points we are using `commit()` and that simplifies a lot of things.

The goal is to be as close as possible to https://github.com/django/django/blob/1.4.22/django/db/transaction.py#L263-L289